### ISSUE TITLE
API to manually dump pgo profiles

### ISSUE THREAD

**Issue by zfernandez88** [oracle/graal#2913]
As far as I can tell, there is no way to generate profile data for profile guided optimizations when building shared objects using native-image. I am able to generate an image using `--pgo-instrument`, but no profile file is created. If I generate an executable, the profile file is created as expected.

If an API is provided, users of shared objects would be able to call the profile dumping API from the appropriate point in their application (eg on shutdown).

I have tested this against 20.2.0 EE (Java8).

**Comment by jramirez-isc** [comment]
Hi @zfernandez88 Thanks for reporting this. 
Currently, we don’t have an API for that. There is only a periodical dump available, but it is a GraalVM EE feature.
The dev team is making an effort to document such an option better. We'll let you know any progress on that.

**Comment by zfernandez88** [comment]
Thanks for looking. I had discussed with @chumer and I did not see the periodic dump option. What version is that flag in? I had tried on EE 20.2.0 (Java8).

Running with -XX:ProfileDumpPeriod yields the following:
<pre>
error: Could not find option 'ProfileDumpPeriod'. Use -XX:PrintFlags= to list all available options.
</pre>
 
If I run with –XX:PrintFlags=, grepping for profile yields the following:
<pre>
  -XX:?InlineMegamorphicCalls                
  -XX:?InlineMonomorphicCalls                
  -XX:?InlinePolymorphicCalls                
                                              
  -XX:?ProfileLockElimination                
  -XX:?ProfileOptBulkAllocation              
  -XX:ProfilesDumpFile="default.iprof"       
  -XX:TruffleInvalidationReprofileCount=3    
  -XX:?TruffleProfilingEnabled               
  -XX:TruffleReplaceReprofileCount=3    
</pre>
